src/lib.rs

```diff
@@ -1043,101 +1043,248 @@ fn graph_k_shortest_path_lengths(
             .collect(),
     })
 }
-/// Return the shortest path lengths between ever pair of nodes that has a
-/// path connecting them
+
+macro_rules! _floyd_warshall {
```

Comment:
Is there a reason you need to use a macro here? It feels like this could just be a common function

Reply:
I did it to avoid code duplication. I tried to make a function with similar same traits to `fn get_edge_iter_with_weights<G>` which accepts both directed and undirected graphs, and it did not work. No trait had `edge_count()` and `node_indices()`.

I think if I split the return statements from the code I can put the logic into functions, but that would still duplicate around 40 lines of code

Reply:
@IvanIsCoding In other functions we do something like:
```rust
fn _floyd_warshall<Ty: EdgeType>(
    py: Python,
    graph: &StableGraph<PyObject, PyObject, Ty>,
    ....
)
```

btw `petgraph::visit::IntoNodeIdentifiers` is the same as `node_indices`